llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -290,37 +291,40 @@ bool SIFoldOperands::updateOperand(FoldCandidate &Fold) const {
 
     if (Fold.Commuted)
       TII->commuteInstruction(*Inst32, false);
-    return true;
-  }
 
-  assert(!Fold.needsShrink() && "not handled");
+    Fold.UseMI = Inst32;
+    Fold.UseOpNo = AMDGPU::getNamedOperandIdx(Fold.UseMI->getOpcode(),
+                                              AMDGPU::OpName::src0);
```

Comment:
Why is that always src0?

Reply:
I see, it must be because a shrunk instruction is VOP1/2/C only and if it is and inline immediate we shall not shrink it. LGTM, but I would add assert here that operand is the use we are looking for.

Reply:
Adding the assert showed up some problems to do with knowing whether or not the instruction has been commuted. I need to spend some more time on that.